client/app.jsx

```diff
@@ -21,9 +26,30 @@ const theme = createTheme({
 });
 
 export default function App() {
+  const [route, setRoute] = useState(parseRoute(window.location.hash));
+
+  window.addEventListener('hashchange', () => {
+    setRoute(parseRoute(window.location.hash));
+  });
```

Comment:
You should set this up in useEffect. Otherwise you are adding a new event listener every single time the app re-renders